mesonbuild/interpreter.py

```diff
@@ -2786,6 +2825,13 @@ def check_subproject_version(wanted, found):
 
     def get_subproject_dep(self, name, dirname, varname, required):
         try:
+            subproject = self.subprojects[dirname]
+            if not subproject.found():
+                if not required:
+                    return subproject
```

Comment:
I'm not sure I like the semantics of returning a SubprojectHolder object here. Shouldn't it be a `NotFoundDependency` wrapped in `DependencyHolder`? Same as `func_dependency()`?

Also, below in `func_dependency()`, inside the `'fallback'` path, the code assumes that any dependency that's found there is usable, which is another argument in favour of returning `NotFoundDependency()` here.

Need a test for this case too, when a fallback subproject was `'auto'`, the dependency was `required: false` or `required: feature-auto`, and the dependency is used in a target.